services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -115,18 +115,26 @@ export class WorkflowService {
       `Context update receivied from client: [${runtimeData.state} -> ${data.state} ]`,
     );
 
-    const updateResult = await this.workflowRuntimeDataRepository.updateById(workflowRuntimeId, {
-      data,
-    });
     // in case current state is a final state, we want to create another machine, of type manual review.
     // assign runtime to user, copy the context.
     const currentState = data.state;
     const workflow = await this.workflowDefinitionRepository.findById(
       runtimeData.workflowDefinitionId,
     );
+
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
-    if (workflow.definition?.states?.[currentState]?.type === 'final') {
+    const isFinal = workflow.definition?.states?.[currentState]?.type === 'final';
+
+    if (isFinal) {
+      data.resolvedAt = new Date();
```

Comment:
Let's not mutate `data`. The line where `data.context` is assigned will also be fixed in the upcoming tech-debt reductions.